src/common/app_metrics.cpp

```diff
@@ -23,7 +23,11 @@ void Buddy::Metrics::RecordRuntimeStats() {
     if (ticks_diff(ticks_ms(), last_recorded_ticks) > 3000) {
         int count = uxTaskGetSystemState(task_statuses, sizeof(task_statuses) / sizeof(task_statuses[1]), NULL);
         for (int idx = 0; idx < count; idx++) {
-            size_t s = malloc_usable_size(task_statuses[idx].pxStackBase);
+            char *stack_base = (char *)task_statuses[idx].pxStackBase;
+            size_t s = 0;
+            /* We can only report free stack space for heap-allocated stack frames. */
+            if (stack_base >= heap_base && stack_base < heap_limit)
+                s = malloc_usable_size(stack_base);
```

Comment:
What do we do with the statically allocated stacks? The metrics provides vital information about stack usage, or better to say - unused stack space which can be reclaimed for some more important stuff.